DotNetHowToAutocomplete/DotNetHowToAutocomplete/Controllers/HomeController.cs

```diff
@@ -0,0 +1,159 @@
+                sp.HighlightPostTag = "</b>";
+            }
+
+            DocumentSuggestResult resp = _indexClient.Documents.Suggest(term, "sg", sp);
+
+            // Convert the suggest query results to a list that can be displayed in the client.
+            List<string> suggestions = resp.Results.Select(x => x.Text).ToList();
+            return new JsonResult
+            {
+                JsonRequestBehavior = JsonRequestBehavior.AllowGet,
+                Data = suggestions
+            };
+        }
+
+        public ActionResult AutoComplete(string term)
+        {
+            InitSearch();
+            //Call autocomplete API and return results
+            AutocompleteParameters sp = new AutocompleteParameters()
+            {
+                AutocompleteMode = AutocompleteMode.OneTermWithContext,
+                UseFuzzyMatching = false,
+                Top = 5,
+                MinimumCoverage = 80
+            };
+            AutocompleteResult resp = _indexClient.Documents.Autocomplete(term, "sg", sp);
+
+            // Conver the Suggest results to a list that can be displayed in the client.
+            List<string> autocomplete = resp.Results.Select(x => x.Text).ToList();
+            return new JsonResult
+            {
+                JsonRequestBehavior = JsonRequestBehavior.AllowGet,
+                Data = autocomplete
+            };
+        }
+
+        public ActionResult Both(string term)
+        {
+            InitSearch();
+
+            AutocompleteParameters sp1 = new AutocompleteParameters()
+            {
+                AutocompleteMode = AutocompleteMode.OneTermWithContext,
+                UseFuzzyMatching = false,
+                Top = 5,
+                MinimumCoverage = 80
+            };
+            AutocompleteResult resp1 = _indexClient.Documents.Autocomplete(term, "sg", sp1);
+
+            // Call both the suggest and autocomplete API and return results
+            SuggestParameters sp2 = new SuggestParameters()
+            {
+                UseFuzzyMatching = false,
+                Top = 5
+            };
+            DocumentSuggestResult resp2 = _indexClient.Documents.Suggest(term, "sg", sp2);
+       
+            //Convert the suggest query results to a list that can be displayed in the client.
+            var result = resp1.Results.Select(x => new { label = x.Text, category = "Autocomplete" }).ToList();
+            result.AddRange(resp2.Results.Select(x => new { label = x.Text, category = "Suggestions" }).ToList());
+
+            return new JsonResult
+            {
+                JsonRequestBehavior = JsonRequestBehavior.AllowGet,
+                Data = result
+            };
+        }
+
+        public ActionResult Facets()
+        {
+            InitSearch();
+
+            // Call suggest API and return results
+            SearchParameters sp = new SearchParameters()
+            {
+                Facets = new List<string> { "agency,count:500" },
```

Comment:
It's good practice to include `Top = 0` in search requests that only need to access facets. This saves a bunch of bandwidth since the actual documents are not returned.